src/read_ext.rs

```diff
@@ -110,6 +110,15 @@ pub trait ReadBinProtExt: io::Read {
         }
         .ok_or(Error::DestinationIntegerOverflow)
     }
+
+    fn bin_read_string(&mut self) -> Result<String> {
+        let len = self.bin_read_nat0::<u64>()? as usize;
```

Comment:
One thing would be to check that serialize_string is encoding the length in Bytes not in Characters. I can't remember which one I did but bytes is the correct one

Reply:
yep, we serialize them as:
```rust
    fn serialize_str(self, v: &amp;str) -&gt; Result&lt;()&gt; {
                self.writer.bin_write_nat0(v.len() as u64)?;
                self.write(v.as_bytes())
            }
```

Reply:
Oh that is wrong then right? v.len() will be the length in characters

Reply:
we want `v.as_bytes().len()`

Reply:
no, https://doc.rust-lang.org/std/primitive.str.html#method.len returns the length in bytes. to get length of characters, we have to use `s.chars().count()`.

Reply:
Oh cool I didn't know that :+1: